app/models/ability.rb

```diff
@@ -0,0 +1,12 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    can :read, :all
+
+    if user.present?
+      can [:manage], CharacterType, user_id: user.id
+      can [:create, :update, :destroy], CharacterAttribute, :character_type => { user_id: user.id }
```

Comment:
@StefanGajic I just realized that these too can be replaced by `:manage`. Read is already granted above.